platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/SingleProblemEvent.java

```diff
@@ -28,6 +28,15 @@
  */
 @Incubating
 public interface SingleProblemEvent extends ProblemEvent {
+
+    /**
+     * Returns the problem definition.
+     *
+     * @return the definition
+     * @since 8.8
+     */
+    ProblemDefinition getDefinition();
+
```

Comment:
@reinsch82 wdyt, should we remove outdated categories like `#getCategory()`? It's a potential breaking change. At the same time, this is an incubating API?

Reply:
I would support removing outdated APIs, we never get rid of it once it's official

Reply:
This PR is getting way too big. I'll merge it and do the cleanup separately.